src/sparseml/pytorch/__init__.py

```diff
@@ -26,6 +26,15 @@
     import torch
 
     _PARSED_TORCH_VERSION = version.parse(torch.__version__)
+
+    if _PARSED_TORCH_VERSION.major >= 2:
+
+        def raise_torch_compile_warning():
+            warnings.warn("torch.compile is not supported by sparseml for torch 2.0.x")
+            return torch.compile
+
+        torch.compile = raise_torch_compile_warning()
```

Comment:
the way that this is setup, it looks like the warning will display on load of this init file since `raise_torch_compile_warning` is invoked (`()`). is this intended? similarly, if the warning is missed `torch.compile(model)` will raise an arguments error